### ISSUE TITLE
Getting exception NoClassDefFoundError 

### ISSUE THREAD

**Issue by davidamir10** [jenkinsci/datadog-plugin#411]
**Describe the bug**
We using jenkins `2.439-alpine-jdk17` on eks with DD plugin. After adding setProperty script to our image, we deployed and got an exception during pipeline run:

> git rev-list --no-walk ...

org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
General error during conversion: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/gitclient/RepositoryCallback

java.lang.RuntimeException: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/gitclient/RepositoryCallback
	at org.codehaus.groovy.control.CompilationUnit.convertUncaughtExceptionToCompilationError(CompilationUnit.java:1123)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1101)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:624)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:602)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:579)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:323)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:293)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox$Scope.parse(GroovySandbox.java:163)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:190)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:175)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:635)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:581)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:335)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:442)
Caused by: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/gitclient/RepositoryCallback
	at org.datadog.jenkins.plugins.datadog.model.BuildData.isGit(BuildData.java:494)
	at org.datadog.jenkins.plugins.datadog.model.BuildData.populateEnvVariables(BuildData.java:359)
	at org.datadog.jenkins.plugins.datadog.model.BuildData.<init>(BuildData.java:156)
	at org.datadog.jenkins.plugins.datadog.listeners.DatadogSCMListener.onCheckout(DatadogSCMListener.java:93)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:182)
	at org.jenkinsci.plugins.workflow.libs.SCMBasedRetriever.lambda$doRetrieve$4(SCMBasedRetriever.java:200)
	at org.jenkinsci.plugins.workflow.libs.SCMBasedRetriever.retrySCMOperation(SCMBasedRetriever.java:222)
	at org.jenkinsci.plugins.workflow.libs.SCMBasedRetriever.doRetrieve(SCMBasedRetriever.java:199)
	at org.jenkinsci.plugins.workflow.libs.SCMSourceRetriever.retrieve(SCMSourceRetriever.java:79)
	at org.jenkinsci.plugins.workflow.libs.LibraryAdder.retrieve(LibraryAdder.java:264)
	at org.jenkinsci.plugins.workflow.libs.LibraryAdder.add(LibraryAdder.java:151)
	at org.jenkinsci.plugins.workflow.libs.LibraryDecorator$1.call(LibraryDecorator.java:125)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1087)
	... 13 more
Caused by: java.lang.ClassNotFoundException: org.jenkinsci.plugins.gitclient.RepositoryCallback
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at jenkins.util.URLClassLoader2.findClass(URLClassLoader2.java:35)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 26 more

1 error

	at org.codehaus.groovy.control.ErrorCollector.failIfErrors(ErrorCollector.java:309)
	at org.codehaus.groovy.control.CompilationUnit.applyToPrimaryClassNodes(CompilationUnit.java:1107)
	at org.codehaus.groovy.control.CompilationUnit.doPhaseOperation(CompilationUnit.java:624)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:602)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:579)
	at groovy.lang.GroovyClassLoader.doParseClass(GroovyClassLoader.java:323)
	at groovy.lang.GroovyClassLoader.parseClass(GroovyClassLoader.java:293)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox$Scope.parse(GroovySandbox.java:163)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.doParse(CpsGroovyShell.java:190)
	at org.jenkinsci.plugins.workflow.cps.CpsGroovyShell.reparse(CpsGroovyShell.java:175)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.parseScript(CpsFlowExecution.java:635)
	at org.jenkinsci.plugins.workflow.cps.CpsFlowExecution.start(CpsFlowExecution.java:581)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:335)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:442)

We then upgraded DD plugin from 5.6.1 to 6.02. This solved the above error, but we are not getting metrics any more. In jenkins log we see exceptions like:

Apr 11, 2024 11:27:08 AM WARNING jenkins.util.Listeners lambda$notify$0
null
java.lang.ClassNotFoundException: org.jenkinsci.plugins.gitclient.RepositoryCallback
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at jenkins.util.URLClassLoader2.findClass(URLClassLoader2.java:35)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
Caused: java.lang.NoClassDefFoundError: org/jenkinsci/plugins/gitclient/RepositoryCallback
	at org.datadog.jenkins.plugins.datadog.model.BuildData.isGit(BuildData.java:512)
	at org.datadog.jenkins.plugins.datadog.model.BuildData.populateEnvVariables(BuildData.java:370)
	at org.datadog.jenkins.plugins.datadog.model.BuildData.<init>(BuildData.java:160)
	at org.datadog.jenkins.plugins.datadog.listeners.DatadogBuildListener.onInitialize(DatadogBuildListener.java:107)
	at hudson.model.listeners.RunListener.lambda$fireInitialize$1(RunListener.java:218)
	at jenkins.util.Listeners.lambda$notify$0(Listeners.java:59)
	at jenkins.util.Listeners.notify(Listeners.java:67)
	at hudson.model.listeners.RunListener.fireInitialize(RunListener.java:216)
	at hudson.model.Run.onStartBuilding(Run.java:2032)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:288)
	at hudson.model.ResourceController.execute(ResourceController.java:101)
	at hudson.model.Executor.run(Executor.java:442)

Label the issue properly.
- Add `severity/` label.
- Add `documentation` label if this issue is related to documentation changes.

**To Reproduce**
This issue observed when we redeployed jenkins base image `2.439-alpine-jdk17` and added setProperty script
The issue started after the controller was up again.

**Expected behavior**
Metrics are sent successfully

**Environment and Versions (please complete the following information):**
A clear and precise description of your setup:
  - Jenkins docker `2.439-alpine-jdk17`
  - kubernetes plugin 4174.v4230d0ccd951
  - DD plugin 5.6.1 (after upgrade 6.0.2). 

**Additional context**
I upgraded jenkins as well as all other plugin to latest, issue still persists.

**Comment by dduportal** [comment]
Same issue appeared on ci.jenkins.io.

The error appeared after upgrading the following plugins today (11 April 2024):

```
aws-java-sdk-ec2
aws-java-sdk-minimal
blueocean-bitbucket-pipeline
blueocean-commons
blueocean-config
blueocean-core-js
blueocean-dashboard
blueocean-events
blueocean-git-pipeline
blueocean-github-pipeline
blueocean-i18n
blueocean-jwt
blueocean-personalization
blueocean-pipeline-api-impl
blueocean-pipeline-editor
blueocean-pipeline-scm-api
blueocean-rest-impl
blueocean-rest
blueocean-web
blueocean
github-branch-source
jenkins-design-language
pipeline-graph-view
pipeline-model-api
pipeline-model-definition
pipeline-model-extensions
pipeline-stage-tags-metadata
script-security
ssh-agent
support-core
```

**Comment by rishiv5** [comment]
Been having the same issue, see #410 
Seems to be resolved in #413 

@nikita-tkachenko-datadog can I ask if you have a timeline for the new version release?

**Comment by nikita-tkachenko-datadog** [comment]
@rishiv5 we are planning to release the fix shortly, most likely in 1-2 hours

**Comment by rishiv5** [comment]
> @rishiv5 we are planning to release the fix shortly, most likely in 1-2 hours

Awesome, thanks!

**Comment by dduportal** [comment]
Thanks @nikita-tkachenko-datadog !

**Comment by nikita-tkachenko-datadog** [comment]
The fix has been released in the plugin v6.0.3. Please try updating and let me know if the issue persists. Thank you!

**Comment by davidamir10** [comment]
@nikita-tkachenko-datadog Thanks for quick response, fix looks good.
